src/hotspot/share/runtime/threadSMR.hpp

```diff
@@ -328,6 +328,10 @@ class ThreadsListHandle : public StackObj {
     return _list_ptr.list();
   }
 
+  using Iterator = ThreadsList::Iterator;
```

Comment:
Is this like a typedef? I don't think I want to see plain "Iterator" and not know what the actual type is here.

Reply:
It's a type alias, which in this simple form is just like a typedef.  Type aliases can also have template parameters.  The iterator type used for iterating over a ThreadsListHandle today happens to be the same type as used for iterating over the underlying ThreadsList.  There's no particular reason to wire that information in to clients.

Reply:
So `ThreadsListHandle::Iterator` is a global(?) alias for `ThreadsList::Iterator`?

If find this name/type aliasing somewhat obscure and confusing. I mean ideally the thing returned would just be an `Iterator` and the details of whether it is a `Foo::Iterator` or `Bar::Iterator` would not be evident in the API.